source/hooks/useApps.jsx

```diff
@@ -27,17 +43,39 @@ const useApps = () => {
   }, [walletNumber]);
 
   useEffect(() => {
+    console.log('apps', apps)
     storage.set({
       [walletNumber]: { apps },
     });
     const parsed = Object.values(apps);
+    console.log('parsed', parsed)
+
+    const allEvents = parsed.flatMap(app =>
+      app.events.map(event => {
+        return {
+          date: event.date,
+          status: event.status,
+          icon: app.icon,
+          name: app.name,
+          url: app.url,
+          whitelist: app.whitelist
+        }
+      })
+    );
+
+    console.log('allEvents', allEvents);
+
     const filtered = parsed.filter((a) => a.status === CONNECTION_STATUS.accepted);
+    const historic = allEvents.filter((a) => a.status === CONNECTION_STATUS.accepted || CONNECTION_STATUS.disconnected);
+
     setParsedApps(filtered);
+    setHistoricApps(historic);
   }, [apps]);
 
   return {
     parsedApps,
     removeApp: handleRemoveApp,
+    historicApps,
```

Comment:
what's parsedApps for ?

Reply:
its parsed + filtered (only accepted) apps